tests/test_parser/test_openapi.py

```diff
@@ -527,57 +528,79 @@ def test__add_parameters_parse_error(self, mocker):
             property_schemas,
         )
 
-    def test__add_parameters_fail_loudly_when_location_not_supported(self, mocker):
-        from openapi_python_client.parser.openapi import Endpoint, Schemas
-
-        endpoint = self.make_endpoint()
+    def test_validation_error_when_location_not_supported(self, mocker):
         parsed_schemas = mocker.MagicMock()
         mocker.patch(f"{MODULE_NAME}.property_from_data", return_value=(mocker.MagicMock(), parsed_schemas))
-        param = oai.Parameter.construct(
-            name="test", required=True, param_schema=mocker.MagicMock(), param_in="error_location"
-        )
-        schemas = Schemas()
-        config = MagicMock()
-
-        result = Endpoint._add_parameters(
-            endpoint=endpoint, data=oai.Operation.construct(parameters=[param]), schemas=schemas, config=config
-        )
-        assert result == (ParseError(data=param, detail="Parameter must be declared in path or query"), parsed_schemas)
+        try:
+            oai.Parameter.construct(
+                name="test", required=True, param_schema=mocker.MagicMock(), param_in="error_location"
+            )
+        except Exception as e:
+            assert e is pydantic.ValidationError
```

Comment:
This test isn't actually checking that the exception happens, only that it's correct if there is one (which there isn't because `construct()` bypasses validation). You'll want something like this:
```suggestion
        with pytest.raises(pydantic.ValidationError):
            oai.Parameter(
                name="test", required=True, param_schema=mocker.MagicMock(), param_in="error_location"
            )
```
Brings up a good point though, we should probably add tests for all the validation code.

[pytest raises docs](https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions)

Reply:
Oh sure. Thank you!